### PR TITLE
New module mail_operating_unit to introduce the operating unit to mail.

### PR DESCRIPTION
**Pull request by BT-ojossen**

### CODE REVIEW

File: mail_operating_unit/wizard/mail_compose_message_view_ext.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    <record model="ir.ui.view" id="email_compose_message_wizard_form_mail">
+        <field name="name">mail.compose.message.form.mail</field>
+        <field name="model">mail.compose.message</field>
+        <field name="inherit_id" ref="mail.email_compose_message_wizard_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='active_domain']" position="after">
+                <field name="operating_unit_id" invisible="1"/>
+            </xpath>
+            <field name="template_id" position="attributes">
```
Comment (BT-rmartin) [inline mail_operating_unit/wizard/mail_compose_message_view_ext.xml:12]: Should this be done by Access Rules? Have a look at product_operating_unit or others. 
The idea is that each operating unit just see what it is assigned on. To be sure about that, it's better to do it with access rules than limiting visibility on each view.

Comment (BT-ojossen) [reply mail_operating_unit/wizard/mail_compose_message_view_ext.xml:12]: strange, I did it like in operating_unit. there it is all done like this...

File: mail_operating_unit/models/mail_mail_ext.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def create(self, vals_list):
+        mails = super(MailMailExt, self).create(vals_list)
+
+        for mail in mails:
+            if mail.mail_message_id.model and \
+                    mail.mail_message_id.res_id:
+
+                model = mail.mail_message_id.model
+                res_id = mail.mail_message_id.res_id
+                record = self.env[model].sudo().browse(res_id)
```
Comment (BT-rmartin) [inline mail_operating_unit/models/mail_mail_ext.py]: Why sudo? We shouldn't break the rules and access rights.

File: mail_operating_unit/models/mail_mail_ext.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                sender_email_address = False
+                if record.operating_unit_id and \
+                        record.operating_unit_id.catchall_alias and \
+                        record.operating_unit_id.catchall_domain:
+                    sender_email_address = \
+                        record.operating_unit_id.catchall_alias + '@' + \
+                        record.operating_unit_id.catchall_domain
+
+                outgoing_mail_server_id = False
+                if record.operating_unit_id:
```
Comment (BT-rmartin) [inline mail_operating_unit/models/mail_mail_ext.py:27]: This condition was already checked in line 28. You could just check once, do an internal if for lines 29-33 and do this assignment

File: mail_operating_unit/models/mail_mail_ext.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                    outgoing_mail_server_id = \
+                        record.operating_unit_id.outgoing_mail_server_id
+
+                if sender_email_address:
+                    email_name = False
+                    if record.operating_unit_id:
```
Comment (BT-rmartin) [inline mail_operating_unit/models/mail_mail_ext.py]: Same as before, already checked. Please, refactor the code

File: mail_operating_unit/models/mail_mail_ext.py (hunk 2 above)
Comment (BT-ojossen) [reply mail_operating_unit/models/mail_mail_ext.py]: This code was coming from Stern. And there Frédéric did not know anymore why he had to do the sudo. I removed it now, so we will see it in the tests if it still works without sudo().

File: mail_operating_unit/models/mail_mail_ext.py (hunk 3 above)
Comment (BT-ojossen) [reply mail_operating_unit/models/mail_mail_ext.py:27]: Done

File: mail_operating_unit/models/mail_mail_ext.py (hunk 4 above)
Comment (BT-ojossen) [reply mail_operating_unit/models/mail_mail_ext.py]: Done

**Pull request merged**